include/zephyr/net/net_if.h

```diff
@@ -368,19 +368,28 @@ struct net_if_dhcpv6 {
 #endif
 /** @endcond */
 
+/**
+ * @brief Network Interface unicast IPv4 address and netmask
+ *
+ * Stores the unicast IPv4 address and related netmask.
+ */
+struct net_if_addr_ipv4 {
+	/** IPv4 address */
+	struct net_if_addr ipv4;
+	/** Netmask */
+	struct in_addr netmask;
+};
```

Comment:
If a netmask is always a bit mask of a certain length `111...00` (which I think it is), you could have the netmask length set in the `_unused` part of the `struct net_if_addr`.

Reply:
This is in line with the `<ip>/<netmask len>` notation, for example `192.168.0.1/24`.

Reply:
We could store the length too, but I went this route so calculating the netmask beforehand and then using it directly. With length, one would need to calculate the netmask for the matching operation. We loose three bytes, netmask is 4 bytes long, vs. one byte for the length when storing the data.

Reply:
It still fits in the unused 5 bits, but more importantly, the interface struct doesn't break with out-of-tree usage.

Reply:
Sure, everything is possible of course, but I suggest that we go with this proposal atm. If you feel strong about this, it is possible to change how this works in subsequent PRs.

Reply:
Not that strongly opiniated, but I'm always cautious when modifying (public) header files. If we accept the proposal no "revert" PR should be added.

Reply:
CC @rlubos

Reply:
I don't have a strong opinion on this, there are certainly various ways/places where netmask could be stored alongside the address, current approach definitely works and seems rather straightforward in use.

But perhaps we should indeed mention this change in the migration guide?

Reply:
> But perhaps we should indeed mention this change in the migration guide?

Yes, that was my plan too. Will do it now when we have a migration doc for 3.7.